Add tests to WKT and WKB conversion to increase coverage

felixdivo opened the pull request:
Adds more meaningful tests for `shapely.wkt` and `shapely.wkb`. They are now covered completely.

felixdivo commented:
I don't know what's up with Shapely 2.0 (and it was not mentioned in the [dev docs](https://shapely.readthedocs.io/en/stable/project.html#development-and-testing)): Is the `master` branch the correct target or would `shapely-2.0` be better suited?

sgillies commented:
@felixdivo master branch, which will be 1.8, is correct. Thank you! I'll have a deeper look tomorrow.

jorisvandenbossche commented:
@felixdivo thanks for the PR! I added a few suggestions

felixdivo commented on tests/test_wkb.py:
```diff
@@ -92,6 +96,64 @@ def test_loads_srid():
 
 
+def test_loads_hex(some_point):
+    assert loads(dumps(some_point, hex=True), hex=True), some_point
```
Yes! It was meant to be like in your suggestion!

felixdivo commented on tests/test_wkb.py:
```diff
@@ -92,6 +96,64 @@ def test_loads_srid():
+
+
+def test_dump_load_binary(some_point):
+    with TemporaryDirectory() as directory_path:
```
Sounds good.

felixdivo commented on tests/test_wkb.py:
```diff
@@ -92,6 +96,64 @@ def test_loads_srid():
+        with open(file, "w") as file_pointer:
+            dump(some_point, file_pointer, hex=True)
+
+        try:
+            with open(file, "rb") as file_pointer:
+                load(file_pointer)
+        except WKBReadingError:
+            pass
+        else:
+            raise AssertionError("Reading a text file as binary should fail")
```
Looks good! I usually use just `unittest` so I didn't know what `pytest` can do ^^

felixdivo commented:
Okay, I added your suggestions and adjusted the rest accordingly. Some test fail, but only on Python 3.5, which is EOL anyway. So is that a problem or is the support for it not dropped already?

felixdivo commented:
According to #886, this release should still support Python 3.5. So, shall I look into the failing tests? I'd rather point it at the 2.0 branch since fixing "bugs" for a platform that is discourage to be used seems like some waste of time, right?

sgillies commented:
I've removed Python 3.5 from the build matrix. I'll merge this and make some post-merge tweaks.
